### DISCUSSION TITLE
Samples wishlist

### DISCUSSION THREAD

**Discussion by krupitskas** [zig-gamedev/zig-gamedev#539] [General]
Hey! I really love the progress and currently also learning Zig. Maybe you have some paticular samples which you would to code, but just haven't time, so I or community could contribute?

**Comment by michal-z** [comment]
Thanks!

Yes of course, such contribution would be very welcome. Also, it is great to hear that you are learning Zig!

Some ideas:
* intro application that renders few different meshes with different orientations (we could show how to use various matrix/quaternion function from zmath lib)
* intro application that renders a lot of small meshes using instancing (transforms would be fetched from large buffer using InstanceID, also we could update transforms each frame on the CPU or GPU)
* intro application that would draw a scene in compute shader using distance field raymarching (shadertoy)
* sample applications that generates and draws simple terrain mesh (we could use stb_perlin lib to generate terrain mesh on the CPU)
* deferred shading sample application (draw Sponza using simple deferred renderer)
* simple 2D fluid simulation in compute shader

Those are the samples I would like to have but feel free to modify/mix them to match your interests.

**Comment by krupitskas** [comment]
I think I pick instancing and add it as an intro 6, thanks.

**Comment by Srekel** [comment]
I made a simple terrain generator/viewer with sokol, aiming to get something similar up and running using zig-gamedev, can try to make a demo using that (though I'm very much not a graphics coder so going to say right now that it won't be... perfect :D ).

(A bit preoccupied with getting something else up and running first so could be a while. Happy to share my sokol project if that would be helpful to anyone else wanting to give it a try).

**Comment by michal-z** [comment]
@Srekel Sounds interesting. Your demo contribution would be very welcome - it doesn't need to be perfect - feel free to submit a PR and we can take if from there.

Regarding your sokol project, I'm afraid I won't be able to help - I haven't used it and I need to focus `zmath v0.3` and `audio_experiments` demo.

**Comment by thomas992** [comment]
I want an ocean with a skybox and a space demo. I think they are great starting points to add game assets too that do not require a terrain editor to be 'finished'

**Comment by michal-z** [comment]
@thomas992 Do you want to contribute?

**Comment by thomas992** [comment]
> @thomas992 Do you want to contribute?

https://github.com/cosinekitty/astronomy/issues/237
I want to have an accurately positioned and scaled solar system, I am going to start converting parts of Astronomy engine to Zig. I could use the C++ library but I rather have the code in Zig. I should be able to draw individual spheres to represent the planets in the game if the library works as I anticipate. I am thinking that my game will use a space skybox and regular BSP type maps, perhaps generated from geographical data. I think that it would be better to be able to zoom around like Google Earth but I do not know how to go about doing something like that.

**Comment by hazeycode** [comment]
A small `zflecs` demo would be cool!
